### PR TITLE
VR > Template data, obj

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/133424

https://www.loom.com/share/635b09f33edc4924a87d24c2049a93dd

### CODE REVIEW

**Comment by paulmskim** [comment]
@juanfra make sure tests are passing.

**Comment by borkweb** [comment]
Why no artifact? The views are being updated/changed?

**Comment by juanfra** [comment]
@borkweb no, we're not changing functionality or how they look. It's only making a pass through the views and using the `$event` object and its properties instead of some functions.

**Comment by borkweb** [comment]
I see code changes in the template files, which are a visible component of our plugins. A change in functionality isn't a prerequisite to needing an artifact. A video showing the non-impactful nature of the changes is valuable for both QA and for you to check your work.

**Comment by borkweb** [comment]
Perhaps some elaboration on my part may be worthwhile. These are a few changes that I see as notable:

* Use of the `$event` object as you mentioned
* How schedule details are output
* The change to how featured event conditionals are handled
* The variable for venue data fetching
* The way start date is put together
* The change on how classes are output (using `tribe_classes()`)

While seemingly minor, a screencast would allow a QA person to look at all of the pieces that should matter (the above items) and also allow you to self-check the success of those changes together in their entirety.

**Comment by juanfra** [comment]
Ok, I can do that. I mean, I did it already, but as we're not changing any behavior and it's something more like using, for example, `$event->featured` instead of `tribe( 'tec.featured_events' )->is_featured( $event_id )` in all views, I didn't recorded any screencast or made any screenshot.

**Comment by juanfra** [comment]
Perfect, will do - thank you Matt! 🎉

File: src/views/v2/day/event/venue.php [hunk 1]
```diff
@@ -9,14 +9,12 @@
...
- * @version 4.9.4
+ * @version TBD
  *
  */
-$event    = $this->get( 'event' );
-$event_id = $event->ID;
 
 // Setup an array of venue details for use later in the template
-$venue_details = tribe_get_venue_details( $event_id );
+$venue_details = tribe_get_venue_details( $event->ID );
```
Comment (paulmskim) [inline src/views/v2/day/event/venue.php]: are we using tribe template tags to get venue details? or should we be getting it from the event object?

File: src/views/v2/list/event/venue.php [hunk 2]
```diff
@@ -9,14 +9,12 @@
...
- * @version 4.9.3
+ * @version TBD
  *
  */
-$event    = $this->get( 'event' );
-$event_id = $event->ID;
 
 // Setup an array of venue details for use later in the template
-$venue_details = tribe_get_venue_details( $event_id );
+$venue_details = tribe_get_venue_details( $event->ID );
```
Comment (paulmskim) [inline src/views/v2/list/event/venue.php]: same question on venue details

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/date.php [hunk 3]
```diff
@@ -42,7 +42,7 @@ class="tribe-events-calendar-month-mobile-events__mobile-event-datetime-featured
 		<span class="tribe-events-calendar-month-mobile-events__mobile-event-datetime-separator">
 			<?php echo esc_html( tribe_get_option( 'timeRangeSeparator', ' - ' ) ); ?>
 		</span>
-		<time datetime="<?php echo esc_attr($event->dates->end->format( 'H:i' )) ?>">
+		<time datetime="<?php echo esc_attr( $event->dates->end->format( 'H:i' ) ) ?>">
```
Comment (paulmskim) [inline src/views/v2/month/mobile-events/mobile-day/mobile-event/date.php]: update template version

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/title.php [hunk 4]
```diff
@@ -26,6 +26,6 @@
 		rel="bookmark"
 		class="tribe-events-calendar-month-mobile-events__mobile-event-title-link tribe-common-anchor"
 	>
-		<?php echo get_the_title( $event ) ?>
+		<?php echo get_the_title( $event->ID ) ?>
```
Comment (paulmskim) [inline src/views/v2/month/mobile-events/mobile-day/mobile-event/title.php]: update template version

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/date.php (hunk 3 above)
Comment (paulmskim) [reply src/views/v2/month/mobile-events/mobile-day/mobile-event/date.php]: never mind, saw that this is already TBD

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/title.php (hunk 4 above)
Comment (paulmskim) [reply src/views/v2/month/mobile-events/mobile-day/mobile-event/title.php]: never mind, saw that this is already TBD

File: src/views/v2/day/event/venue.php (hunk 1 above)
Comment (juanfra) [reply src/views/v2/day/event/venue.php]: Right now we don't have any other way of getting them besides building something similar to what the template tag does, but in the template.

@bordoni thoughts?

**Comment by juanfra** [comment]
@paulmskim can I please have a new review on this one? Thanks!

**Pull request merged**